### DISCUSSION TITLE
Why using multiset instead of collections.Counter?

### DISCUSSION THREAD

**Discussion by michelececcacci** [sympy/sympy#24678] [Q&A] [answered]
`collections.Counter` does the same thing as the `multiset` function in `sympy/utilities/iterables.py`. Are there any specific reasons to not use `Counter`?

**Comment by sylee957** [comment, marked as answer]
`collections.Counter` was introduced after python 3 and sympy had historically existed from python 2.
https://docs.python.org/3/library/collections.html#collections.Counter
We can deprecate such old function or guide users against using it in documentations.

**Reply by michelececcacci** [reply]
Ok, perfect, thanks a lot. If you need help replacing all the occurrences of the function in the codebase i can do it!

**Comment by asmeurer** [comment]
I agree we should deprecate this function in favor of `dict(collections.Counter)`. 

It might be interesting to add a true symbolic mathematical multiset to the sets module. I don't think any of the existing uses of `multiset` necessarily need this, though.

**Reply by michelececcacci** [reply]
Do you mind if i start replacing occurrences in the codebase?

**Reply by asmeurer** [reply]
Go ahead. If you want to add a deprecation, you can do that as well. See https://docs.sympy.org/latest/contributing/deprecations.html

**Reply by michelececcacci** [reply]
Thanks a lot, i will check it out and start working on it.
